src/rest_api.py

```diff
@@ -68,29 +81,144 @@ def recommender():
+              method=request.method,
+              status_code=status)
 
     return flask.jsonify(r), status
 
 
 @app.route('/api/v1/stack_aggregator', methods=['POST'])
-def stack_aggregator():
+def stack_aggregator(test=False, request=request):
     """Handle POST requests that are sent to /api/v1/stack_aggregator REST API endpoint."""
     s = {'stack_aggregator': 'failure', 'external_request_id': None}
     input_json = request.get_json()
-    current_app.logger.debug('stack_aggregator/ request with payload: {p}'.format(p=input_json))
-    if input_json and 'external_request_id' in input_json and input_json['external_request_id']:
+
+    if not test:
+        current_app.logger.debug('stack_aggregator/ request with payload: {p}'.format(p=input_json))
+
+    payload = {
+        "pid": os.getpid(),
+        "hostname": os.environ.get("HOSTNAME"),
+        "endpoint": request.endpoint,
+        "method": request.method
+    }
+    if input_json and 'external_request_id' in input_json \
+            and input_json['external_request_id']:
+
         try:
             persist = request.args.get('persist', 'true') == 'true'
             s = StackAggregator().execute(input_json, persist=persist)
+            status_code = 200
+            if s is not None:
+                total_time_elapsed(
+                    sa_audit_data=s['result']['_audit'],
+                    external_request_id=input_json['external_request_id'],
+                    worker=stack_analysis_workers[1])
+
         except Exception as e:
             s = {
                 'stack_aggregator': 'unexpected error',
                 'external_request_id': input_json.get('external_request_id'),
                 'message': '%s' % e
             }
+            status_code = 400
+        push_data(data=s, status_code=status_code, **payload)
+
+    if test:
+        return s
 
     return flask.jsonify(s)
 
 
+def total_time_elapsed(
+        sa_audit_data,
+        external_request_id,
+        worker, url=METRICS_COLLECTION_URL, rq=request):
+    """
+     Pushes Combined Metrics data to specified Url. Called in Stack Aggregator Only.
+
+    :param: sa_audit_data: Stack Aggregator Audit Data
+    :param: external_request_id: Stack Id
+    :param: worker: recommender_v2
+    :return: Metrics Data
+    """
+    input_json = rq.get_json()
+    # current_app.logger.debug('total_time_elapsed/ request with payload: {p}'.format(p=input_json))
+    payload = {
+        "pid": os.getpid(),
+        "hostname": os.environ.get("HOSTNAME"),
+        "endpoint": rq.endpoint,
+        "request_method": rq.method,
+    }
+
+    re_db_data = select_from_db(
+                    external_request_id=input_json["external_request_id"], worker=worker)
+
+    if (re_db_data is None) and (sa_audit_data is None):
+        current_app.logger.debug('No Data found wrt Stack Id : {p}'.format(p=external_request_id))
+        payload["value"] = 0
+        payload["status_code"] = 400
```

Comment:
Status cannot be 400 just because other task is not complete. IN this case, you either retry or pass without calling push_data